packages/flutter/lib/src/widgets/view.dart

```diff
@@ -106,13 +142,578 @@ class View extends StatelessWidget {
+    renderObject.child = child;
+  }
+
+  @override
+  void moveRenderObjectChild(RenderObject child, Object? oldSlot, Object? newSlot) {
+    assert(false);
+  }
+
+  @override
+  void removeRenderObjectChild(RenderObject child, Object? slot) {
+    assert(slot == null);
+    assert(renderObject.child == child);
+    renderObject.child = null;
+  }
+
+  @override
+  void unmount() {
+    if (_effectivePipelineOwner != (widget as _RawView)._deprecatedPipelineOwner) {
+      _effectivePipelineOwner.dispose();
+    }
+    super.unmount();
+  }
 }
 
 class _ViewScope extends InheritedWidget {
   const _ViewScope({required this.view, required super.child});
 
-  final FlutterView view;
+  final FlutterView? view;
 
   @override
   bool updateShouldNotify(_ViewScope oldWidget) => view != oldWidget.view;
 }
+
+class _PipelineOwnerScope extends InheritedWidget {
+  const _PipelineOwnerScope({
+    required this.pipelineOwner,
+    required super.child,
+  });
+
+  final PipelineOwner pipelineOwner;
+
+  @override
+  bool updateShouldNotify(_PipelineOwnerScope oldWidget) => pipelineOwner != oldWidget.pipelineOwner;
+}
+
+class _MultiChildComponentWidget extends Widget {
+  const _MultiChildComponentWidget({
+    super.key,
+    List<Widget> views = const <Widget>[],
+    Widget? child,
+  }) : _views = views, _child = child;
+
+  // It is up to the subclasses to make the relevant properties public.
+  final List<Widget> _views;
+  final Widget? _child;
+
+  @override
+  Element createElement() => _MultiChildComponentElement(this);
+}
+
+/// A collection of sibling [View]s.
+///
+/// This widget can only be used in places were a [View] widget is allowed. In
+/// practical terms, it can be used at the root of the widget tree outside of
+/// any [View] widget, as a child to a another [ViewCollection], or in the
+/// [ViewAnchor.view] slot of a [ViewAnchor] widget. It is not required to be a
+/// direct child of those widgets; other non-[RenderObjectWidget]s may appear
+/// in between the two (such as an [InheritedWidget]).
+///
+/// Similarly, the [views] children of this widget must be [View]s, but they
+/// may be wrapped in additional non-[RenderObjectWidget]s (e.g.
+/// [InheritedWidget]s).
+class ViewCollection extends _MultiChildComponentWidget {
+  /// Creates a [ViewCollection] widget.
+  ///
+  /// The provided list of [views] must contain at least one widget.
+  const ViewCollection({super.key, required super.views}) : assert(views.length > 0);
+
+  /// The [View] descendants of this widget.
+  ///
+  /// The [View]s may be wrapped in other non-[RenderObjectWidget]s (e.g.
+  /// [InheritedWidget]s). However, no [RenderObjectWidget] is allowed to appear
+  /// between the [ViewCollection] and the next [View] widget.
+  List<Widget> get views => _views;
+}
+
+/// Decorates a [child] widget in a surrounding [View] with a side-[View].
+///
+/// This widget must have a [View] ancestor, into which the [child] widget
+/// is rendered.
```

Comment:
```suggestion
/// This widget is not used to spawn the root view. This widget looks up a
/// [View] ancestor, and renders the [child] widget into it.
```

Reply:
Hm... it doesn't really look up the View. If we want to get overly technical (which I think this doc doesn't have to do), there has to be an ancestor render object to which the render object of the child can be attached to. In other words, there needs to be some ancestor that creates a render tree, and that ancestor is typically a [View].

Reply:
Ok. I don't want to get overly technical either. What about:
```dart
/// This widget is not used to spawn the root view. This widget renders the
/// [child] widget into an existing [View] ancester.
```

I'm proposing this change because it isn't immediately obviously why we want to emphasize "this widget must have a View ancestor" here. Doesn't anywhere in the widget tree have a view ancestor? _Oh right, the root doesn't._ Well, we might as well just explain this out explicitly.

Reply:
I reworded this a little bit to remove the focus of the ancestor view from the first line summary.